docs/book/src/quinn/certificate.md

```diff
@@ -0,0 +1,192 @@
+    Ok((cert, key))
+}
+
+pub fn parse_pem(cert: Vec<u8>, private_key: Vec<u8>) -> anyhow::Result<(quinn::Certificate, quinn::PrivateKey)> {
+    // Parse to certificate chain whereafter taking the first certifcater in this chain.
+    let cert = quinn::CertificateChain::from_pem(&cert)?.iter().next().unwrap().clone();
+    let key = quinn::PrivateKey::from_pem(&private_key)?;
+
+    Ok((quinn::Certificate::from(cert), key))
+}
+```
+
+There are two common certificate formats namely: `.pem` and `.der`.
+The `.der` certificates are byte-coded, while `.pem` is text-coded.
+You can translate one to the other by using tooling such as openssl or even within code self. 
+The code translation is shown above. 
+
+### Self Signed
+
+A [self-signed][5] certificate entails that you sign a certificate with your own CA. 
+These certificates are easy to create and cost no money. 
+However, they do not offer all the security features that certificates from a CA do have. 
+Some ways to create a self-signed certificate is by using [rcgen][4] or openssl. 
+In this example [rcgen][4] is used.   
+
+Let's look at an example:
+
+```rust
+pub fn generate_self_signed_cert(cert_path: &str, key_path: &str) -> anyhow::Result<(quinn::Certificate, quinn::PrivateKey)> {
+    // Generate dummy certificate.
+    let certificate = rcgen::generate_simple_self_signed(vec!["localhost".into()]).unwrap();
+    let serialized_key = certificate.serialize_private_key_der();
+    let serialized_certificate = certificate.serialize_der().unwrap();
+
+    // Write to files.
+    fs::write(&cert_path, &serialized_certificate).context("failed to write certificate")?;
+    fs::write(&key_path, &serialized_key).context("failed to write private key")?;
+
+    parse_der(serialized_certificate, serialized_key)
+}
+```
+
+*Note that [generate_simple_self_signed][generate_simple_self_signed] returns a [Certificate][2] that can be serialized to both `.der` and `.pem` formats.*
+
+### Official Certificates
+
+[Let's Encrypt][6] is a CA and distributes certificates for free. 
+Its a very well-known CA used by many applications around the world.
+We can cover a detailed lets-encrypt tutorial but there is plenty of good documentation out there.  
+
+**Generate Certificate**
+
+Let's Encrypt works with [Certbot][7], certbort generates the certificate for you.
+Often a certificate is generated to secure a web server. 
+Because we generate a certificate for a protocol, the configuration process will be slightly different than normal. 
+We assume that you do not have a web server. 
+Select on the certbot website that you do not have a web server and follow the given installation instructions.
+
+If certbot is installed, execute `certbot certonly --standalone`, this command will fire up a web server in the background.
+Certbot asks for your data, after entering it two `.pem` files are generated, namely `cert.pem` and `privkey.pem`. 
+Next we can reference those files in the code.  
+ 
```

Comment:
How does this actually work? Seems to gloss over what kind of environment is needed for this to work.

Reply:
Above I mention: _Select on the certbot website that you do not have a web server and follow the given installation instructions._

Certbot website explains this in detail. I thought I would be unnecessarily to cover platform-specific notes since there is _plenty of good documentation to there_

Reply:
I rewrote this section a bit and pointed more to the existing certbot docs

Let's Encrypt use [Certbot][7] to generate certificates. 
The certbot websites provides clear instructions on how to use the tool.   
Normally a certificate is generated to secure a web server and the user is expected to have one up-and-running accessible from a public DNS.
However, since we generate a certificate for a protocol, the configuration process will be slightly different than normal.
If you do want to use an existing web server to generate certificates, please follow the instructions on the certbot website.

For this example it is expected that no web server is installed.
 Select on the certbot web site that you do not have a web server and follow the given installation instructions. 

If certbot is installed, run `certbot certonly --standalone`, this command will start a web server in the background.
Certbot asks for the required data and writes the certificate to `cert.pem` and private key to `privkey.pem`.  
These files can then be referred to in code.

Reply:
Please give it another go in the following review round.